[AIRFLOW-4858] Deprecate "Historical convenience functions" in conf

haoliang7 opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4858
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
The first argument of zope.deprecation.deprecated should be string or string list to deprecate top level function in a module.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

Commit:
Deprecate "Historical convenience functions" in conf

ashb reviewed (changes requested):
Can you please also fix up the existing uses in side airflow that this now correctly warns about?

For example a non-exhaustive list:

- airflow/config_templates/airflow_local_settings.py contains many
- airflow/lineage/__init__.py
- airflow/logging_config.py

haoliang7 committed:
[AIRFLOW-4858] Deprecate "Historical convenience functions" in conf

1. Issue old conf method deprecation warnings properly and remove current old conf method usages.
2. Unify the way to use conf as
from airflow.configuration import conf
3. Fix improper test mock in
tests/jobs/test_scheduler_job.py
tests/test_logging_config.py
If we remove the old conf method, they will mock the conf method and call the same method inside the mock object which will incur maximum recursion error.

ashb commented on airflow/configuration.py:
```diff
@@ -38,7 +38,7 @@
 # show Airflow's deprecation warnings
 warnings.filterwarnings(
-    action='default', category=DeprecationWarning, module='airflow')
+    action='always', category=DeprecationWarning, module='airflow')
```
Why this change? And what effect does this change have?

haoliang7 replied:
This was for testing and accidentally added to the commit. I have changed it back. Sorry for the confusion.
